pkg/service/handlers/v1-kube-api-authn.go

```diff
@@ -97,58 +100,58 @@ func v1getBodyAuthnRequest(bytes []byte) (*types.V1AuthnRequest, error) {
 }
 
 func (h *KubeAPIHandlers) v1getAndVerifyUser(accessKey, secretKey string) (*types.V1AuthnResponseUser, error) {
-	clusterAuthToken, err := h.clusterAuthTokensLister.Get(h.namespace, accessKey)
-	if err != nil {
-		return nil, err
-	}
-	if !clusterAuthToken.Enabled {
-		return nil, fmt.Errorf("token is not enabled")
-	}
-
-	userName := clusterAuthToken.UserName
-	clusterUserAttribute, err := h.clusterUserAttributeLister.Get(h.namespace, userName)
-	if err != nil {
-		return nil, err
-	}
-	if !clusterUserAttribute.Enabled {
-		return nil, fmt.Errorf("user is not enabled")
-	}
-
-	// An is-not-found error is ok. Likely a not-yet-migrated cluster auth token.
-	// Everything else is reported.
-	clusterAuthTokenSecret, err := h.secretLister.Get(h.namespace, common.ClusterAuthTokenSecretName(accessKey))
-	if err != nil && !apierrors.IsNotFound(err) {
-		return nil, err
-	}
+	var (
+		clusterAuthToken       *clusterv3.ClusterAuthToken
+		clusterAuthTokenSecret *corev1.Secret
+		clusterUserAttribute   *clusterv3.ClusterUserAttribute
+	)
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		var err error
+		clusterAuthToken, clusterAuthTokenSecret, clusterUserAttribute, err = h.v1GetClusterAuth(accessKey)
+		if err != nil {
+			return err
+		}
```

Comment:
TBH not a big fan of indirects. I'd rather read top-down what we're doing here instead jumping between methods.

Reply:
reworked, commit a0fdbf0f6065ae35ca34961264d9617f35772523

Me, on the other hand,  not a fan of long functions which do not fit fully on the screen.